chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/AddCuresUpdateToListingsJob.java

```diff
@@ -0,0 +1,127 @@
+
+import java.util.Date;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.transaction.TransactionDefinition;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.TransactionCallbackWithoutResult;
+import org.springframework.transaction.support.TransactionTemplate;
+import org.springframework.web.context.support.SpringBeanAutowiringSupport;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import gov.healthit.chpl.SpecialProperties;
+import gov.healthit.chpl.dao.CertifiedProductDAO;
+import gov.healthit.chpl.dao.CuresUpdateEventDAO;
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.domain.activity.ActivityConcept;
+import gov.healthit.chpl.dto.CertifiedProductDetailsDTO;
+import gov.healthit.chpl.dto.CuresUpdateEventDTO;
+import gov.healthit.chpl.exception.EntityCreationException;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.manager.ActivityManager;
+import gov.healthit.chpl.manager.CertifiedProductDetailsManager;
+import gov.healthit.chpl.service.CuresUpdateService;
+
+public class AddCuresUpdateToListingsJob extends QuartzJob {
+    private static final Logger LOGGER = LogManager.getLogger("addCuresUpdateToListingsJobLogger");
+
+    @Autowired
+    private CuresUpdateService curesUpdateService;
+
+    @Autowired
+    private CertifiedProductDetailsManager certifiedProductDetailsManager;
+
+    @Autowired
+    private CertifiedProductDAO certifiedProductDAO;
+
+    @Autowired
+    private SpecialProperties specialProperties;
+
+    @Autowired
+    private CuresUpdateEventDAO curesUpdateDao;
+
+    @Autowired
+    private ActivityManager activityManager;
+
+    @Autowired
+    private JpaTransactionManager txManager;
+
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
+        LOGGER.info("********* Starting the Update Listing To Cures Updated job. *********");
+
+        getAll2015Listings().stream()
+                .forEach(dto -> processListing(dto));
+
+        LOGGER.info("********* Completed the Update Listing To Cures Updated  job. *********");
+
+    }
+
+    private void processListing(CertifiedProductDetailsDTO cp) {
+        try {
+            LOGGER.info("Retrieving listing information for  (" + cp.getId() + ") " + cp.getChplProductNumber());
+            CertifiedProductSearchDetails listing = certifiedProductDetailsManager.getCertifiedProductDetails(cp.getId());
+            if (curesUpdateService.isCuresUpdate(listing)) {
```

Comment:
This job needs to add a line of "cures update = false" to the cures_update_event table for listings that don't meet the curesUpdate requirement, as well as the "cures update = true" that it's currently doing

It _shouldn't_ add activity for listings that are "cures update = false" though; only for ones that are getting set as true